File: backend/Node.JS/calindra-desafio/src/controllers/v1.0/geolocation/models/responses/geolocation-references-info.ts
import { Exclude } from 'class-transformer';

import { GeolocationInfo } from './geolocation-info';
import { Location } from './location';

export class GeolocationReferencesInfo {
  public placeId: string;
  public address: string;
  public location: Location;
  public greatherDistances: GeolocationInfo[];
  public shorterDistances: GeolocationInfo[];

  @Exclude()
  public addresses: GeolocationInfo[];
}

File: backend/.NET/Calindra.Desafio/Calindra.Desafio.Domain/Models/Responses/GeolocationReferencesInfo.cs
using System.Collections.Generic;
using Calindra.Desafio.Domain.Models.Responses;
using Newtonsoft.Json;

namespace Calindra.Desafio.Domain.Models.Responses
{
    public class GeolocationReferencesInfo
    {
        public GeolocationReferencesInfo()
        {
            Addresses = new List<GeolocationInfo>();
        }

        public string PlaceId { get; set; }
        public string Address { get; set; }
        public Location Location { get; set; }
        public IList<GeolocationInfo> GreatherDistances { get; set; }
        public IList<GeolocationInfo> ShorterDistances { get; set; }

        [JsonIgnore]
        public IList<GeolocationInfo> Addresses { get; set; }           
    }
}
